### ISSUE TITLE
How to make siganature-pad totally optional on a form

### ISSUE THREAD

**Issue by radman63** [thread-pond/signature-pad#182]
Whenever I've used your signature pad on forms, the signature was always required. On the form I'm creating now, the signature is only required if the person is waiving medical attention. I don't see a way to make the signature optional so the form will submit without it. I see there's a way to make the typed-in name optional, which I never use, but I don't see a way to make the whole thing optional. I tried hiding the signature in a hidden div that only gets called if someone checks a checkbox, but that didn't work.

Is there a way to display the signature pad, but make it (and the typed in name both) optional? Thanks.

Regards,
Bill

**Comment by thomasjbradley** [comment]
Yep it's possible to do that! You can use the `validateFields` option and set it to `false`.

**Comment by radman63** [comment]
Thank you for your reply. I set the option as you suggested, which allowed me to submit the form without a signature, but then I received an error on the display side. It worked before I set this option. Maybe I set the option incorrectly?
 
The only thing I changed was adding the line regarding validateFields on the submit form. I need one form to submit the data, and another form to display it.
 
SUBMIT FORM:
 <script>
 $(document).ready(function() {
 var options = {
 bgColour : '#f0f0f5',
 drawOnly : true,
 penColour: '#0087bb',
 lineTop: 65,
 validateFields: false
 };
 $('.sigPad').signaturePad(options);
 });
 </script>
  
 DISPLAY FORM:
 <script>
 $(document).ready(function(){
var options = {
 bgColour : '#ffffff',
 drawOnly : true,
 penColour: '#0087bb',
 lineTop: 65,
 displayOnly: false
 };
 var sig = $$output$$;
 $('.sigPad').signaturePad(options).regenerate(sig)
 })
 </script>
  
 Here's the error I received on the display side:
 
 
 
 
 
--------- Original Message --------- Subject: Re: [thomasjbradley/signature-pad] How to make siganature-pad totally optional on a form (#182)

**Comment by thomasjbradley** [comment]
Can you send me the error you received again? It didn't show up in the message.

**Comment by radman63** [comment]
I was afraid the screenshot wouldn't work. Thanks for your help with this.
 
Line: 18
Char: 14
Error: Syntax error
Code: 0
 
Probably referring to the display script instead of the input script, but I'm not sure, since they are both related. If the input script is bad, it could throw off the display.
 
--------- Original Message --------- Subject: Re: [thomasjbradley/signature-pad] How to make siganature-pad totally optional on a form (#182)

**Comment by thomasjbradley** [comment]
What about the line `var sig = $$output$$`? Does that get populated with something?

**Comment by radman63** [comment]
I think it gets populated with the signature, so maybe there's a problem when that is empty (null)?
 
--------- Original Message --------- Subject: Re: [thomasjbradley/signature-pad] How to make siganature-pad totally optional on a form (#182)

**Comment by thomasjbradley** [comment]
Yeah, that could be a possibility. Maybe check the source in the browser to see if something is there.

**Comment by radman63** [comment]
Unless I'm totally misunderstanding how this works, there wouldn't be anything anywhere, because in this scenario, no one has signed the form. So, there's nothing in that element. 
 
--------- Original Message --------- Subject: Re: [thomasjbradley/signature-pad] How to make siganature-pad totally optional on a form (#182)

**Comment by thomasjbradley** [comment]
So the displayed signature and the signing box are on the same page at the same time?

**Comment by radman63** [comment]
No. They are not on the same page. Let me explain further. You helped me out in this environment before. I am using FileBound Document Management System, and they use your signature pad. The way it works is that I have to create what you might call an external form to capture data (including the signature) and write it to their database. This is the only form that actually gets submitted.
 
Then, I have to create another form to display the data in their system. I spend more time on the external form and try to make it look pretty. The internal form is just a simple display of the data captured.
 
There are other elements on the form, but here are the signature parts of both forms. I'm not sure if the "validateFields: false" needs to be on both forms, or just the external one that gets submitted.
 
EXTERNAL DATA GATHERING FORM 
<script>
$(document).ready(function() {
var options = {
bgColour : '#f0f0f5',
drawOnly : true,
penColour: '#0087bb',
lineTop: 65,
validateFields: false
};
$('.sigPad').signaturePad(options);
});
</script>
 
<form action="html2eform.ashx" method="post" name="eForm" id="eForm" class="sigPad form-horizontal">
 <div class="form-group">
 <div class="col-md-6">
 <p><canvas class="pad" width="350" height="96"></canvas></p>
 <p class="clearButton btn btn-info"><a href="#clear" style="text-decoration: none; color: #EEEEEE;">Clear Signature</a></p>
 <p class="error"></p>
 </div>
 </div>
   <input type="hidden" name="output" class="output">
 INTERNAL DATA DISPLAYING FORM 
 Doesn't even have to be an actual form. It just pulls in the external form elements as variables and displays them, along with static text.
 <script>
 $(document).ready(function(){
var options = {
 bgColour : '#ffffff',
 drawOnly : true,
 penColour: '#0087bb',
 lineTop: 65,
 };
 var sig = $$output$$;
 $('.sigPad').signaturePad(options).regenerate(sig)
 })
 </script>
 
<label for="signature">Injured Person's Signature:</label>
 <form action="$$URL$$" method="post" name="eForm" id="eForm" class="sigPad">
 <canvas class="pad" width="350" height="96" style="left: 0px; top: 0px"></canvas>
 That's it. Thanks again for your help.
 
--------- Original Message --------- Subject: Re: [thomasjbradley/signature-pad] How to make siganature-pad totally optional on a form (#182)

**Comment by thomasjbradley** [comment]
Hmm... I dunno. Are you still getting the error on line 14? Does it point to your code or to signature pad?

**Comment by radman63** [comment]
I'm still getting an error. The only thing I've tried so far is adding and removing "validateFields: False" from the internal display form. The error is on this line, "var sig = $$output$$;", so I'm not sure what the problem is, except that maybe it doesn't like the output to be blank? I posted a request for help on the FileBound forum page also, but have not received a reply yet. I don't know if I'm going to have to create some script to hide your script if the output is blank, or if a completely different solution is required. This is beyond me.
--------- Original Message --------- Subject: Re: [thomasjbradley/signature-pad] How to make siganature-pad totally optional on a form (#182)

**Comment by thomasjbradley** [comment]
Oh, the output being blank is really interesting—that could be it. Maybe try a blank output in the example files and see if the same error is shown.

**Comment by radman63** [comment]
Here is what I see when I view source on the display page.
 
If there is a signature drawn, I see this:
<script>
 $(document).ready(function(){
var options = {
 bgColour : '#ffffff',
 drawOnly : true,
 penColour: '#0087bb',
 lineTop: 65,
 };
 var sig = [{"lx":14,"ly":40,"mx":14,"my":39},{"lx":14,"ly":38,"mx":14,"my":40},{"lx":13,"ly":37,"mx":14,"my":38},{"lx":11,"ly":37,"mx":13,"my":37},{"lx":10,"ly":36,"mx":11,"my":37},{"lx":8,"ly":36,"mx":10,"my":36},{"lx":8,"ly":39,"mx":8,"my":36},{"lx":8,"ly":42,"mx":8,"my":39},{"lx":8,"ly":45,"mx":8,"my":42},{"lx":8,"ly":47,"mx":8,"my":45},{"lx":8,"ly":50,"mx":8,"my":47},{"lx":12,"ly":51,"mx":8,"my":50},{"lx":20,"ly":51,"mx":12,"my":51},{"lx":28,"ly":51,"mx":20,"my":51},{"lx":38,"ly":51,"mx":28,"my":51},{"lx":48,"ly":50,"mx":38,"my":51},{"lx":55,"ly":47,"mx":48,"my":50},{"lx":64,"ly":43,"mx":55,"my":47},{"lx":70,"ly":38,"mx":64,"my":43},{"lx":75,"ly":35,"mx":70,"my":38},{"lx":78,"ly":32,"mx":75,"my":35},{"lx":80,"ly":30,"mx":78,"my":32},{"lx":81,"ly":29,"mx":80,"my":30},{"lx":81,"ly":28,"mx":81,"my":29},{"lx":81,"ly":27,"mx":81,"my":28},{"lx":81,"ly":26,"mx":81,"my":27},{"lx":80,"ly":26,"mx":81,"my":26},{"lx":78,"ly":26,"mx":80,"my":26},{"lx":76,"ly":26,"mx":78,"my":26},{"lx":73,"ly":28,"mx":76,"my":26},{"lx":69,"ly":31,"mx":73,"my":28},{"lx":66,"ly":34,"mx":69,"my":31},{"lx":64,"ly":37,"mx":66,"my":34},{"lx":62,"ly":40,"mx":64,"my":37},{"lx":62,"ly":42,"mx":62,"my":40},{"lx":62,"ly":43,"mx":62,"my":42},{"lx":62,"ly":44,"mx":62,"my":43},{"lx":65,"ly":44,"mx":62,"my":44},{"lx":74,"ly":43,"mx":65,"my":44},{"lx":81,"ly":40,"mx":74,"my":43},{"lx":90,"ly":36,"mx":81,"my":40},{"lx":97,"ly":32,"mx":90,"my":36},{"lx":106,"ly":30,"mx":97,"my":32},{"lx":110,"ly":27,"mx":106,"my":30},{"lx":112,"ly":26,"mx":110,"my":27},{"lx":115,"ly":24,"mx":112,"my":26},{"lx":116,"ly":24,"mx":115,"my":24},{"lx":116,"ly":26,"mx":116,"my":24},{"lx":116,"ly":28,"mx":116,"my":26},{"lx":116,"ly":30,"mx":116,"my":28},{"lx":116,"ly":32,"mx":116,"my":30},{"lx":117,"ly":34,"mx":116,"my":32},{"lx":119,"ly":35,"mx":117,"my":34},{"lx":124,"ly":36,"mx":119,"my":35},{"lx":127,"ly":37,"mx":124,"my":36},{"lx":131,"ly":38,"mx":127,"my":37},{"lx":136,"ly":39,"mx":131,"my":38},{"lx":138,"ly":40,"mx":136,"my":39},{"lx":139,"ly":40,"mx":138,"my":40},{"lx":140,"ly":41,"mx":139,"my":40},{"lx":141,"ly":42,"mx":140,"my":41},{"lx":142,"ly":42,"mx":141,"my":42},{"lx":143,"ly":43,"mx":142,"my":42},{"lx":144,"ly":44,"mx":143,"my":43},{"lx":145,"ly":45,"mx":144,"my":44},{"lx":146,"ly":45,"mx":145,"my":45},{"lx":147,"ly":47,"mx":146,"my":45},{"lx":150,"ly":47,"mx":147,"my":47},{"lx":155,"ly":47,"mx":150,"my":47},{"lx":162,"ly":47,"mx":155,"my":47},{"lx":173,"ly":47,"mx":162,"my":47},{"lx":184,"ly":45,"mx":173,"my":47},{"lx":198,"ly":41,"mx":184,"my":45},{"lx":209,"ly":38,"mx":198,"my":41},{"lx":221,"ly":34,"mx":209,"my":38},{"lx":228,"ly":33,"mx":221,"my":34},{"lx":232,"ly":31,"mx":228,"my":33},{"lx":233,"ly":31,"mx":232,"my":31},{"lx":234,"ly":31,"mx":233,"my":31},{"lx":233,"ly":35,"mx":234,"my":31},{"lx":231,"ly":39,"mx":233,"my":35},{"lx":230,"ly":42,"mx":231,"my":39},{"lx":229,"ly":43,"mx":230,"my":42},{"lx":229,"ly":44,"mx":229,"my":43},{"lx":229,"ly":45,"mx":229,"my":44},{"lx":229,"ly":44,"mx":229,"my":45},{"lx":229,"ly":43,"mx":229,"my":44},{"lx":228,"ly":43,"mx":229,"my":43},{"lx":227,"ly":43,"mx":228,"my":43},{"lx":223,"ly":42,"mx":227,"my":43},{"lx":219,"ly":42,"mx":223,"my":42},{"lx":214,"ly":42,"mx":219,"my":42},{"lx":213,"ly":42,"mx":214,"my":42},{"lx":221,"ly":42,"mx":213,"my":42},{"lx":238,"ly":42,"mx":221,"my":42},{"lx":260,"ly":42,"mx":238,"my":42},{"lx":288,"ly":42,"mx":260,"my":42},{"lx":310,"ly":42,"mx":288,"my":42},{"lx":327,"ly":42,"mx":310,"my":42},{"lx":341,"ly":42,"mx":327,"my":42},{"lx":345,"ly":42,"mx":341,"my":42},{"lx":345,"ly":43,"mx":345,"my":42},{"lx":342,"ly":44,"mx":345,"my":43},{"lx":335,"ly":45,"mx":342,"my":44},{"lx":326,"ly":48,"mx":335,"my":45},{"lx":315,"ly":49,"mx":326,"my":48},{"lx":303,"ly":51,"mx":315,"my":49},{"lx":293,"ly":55,"mx":303,"my":51},{"lx":287,"ly":56,"mx":293,"my":55},{"lx":280,"ly":58,"mx":287,"my":56},{"lx":279,"ly":58,"mx":280,"my":58},{"lx":277,"ly":58,"mx":279,"my":58},{"lx":274,"ly":57,"mx":277,"my":58},{"lx":270,"ly":54,"mx":274,"my":57},{"lx":263,"ly":52,"mx":270,"my":54},{"lx":257,"ly":49,"mx":263,"my":52},{"lx":251,"ly":48,"mx":257,"my":49},{"lx":248,"ly":48,"mx":251,"my":48},{"lx":245,"ly":47,"mx":248,"my":48},{"lx":244,"ly":47,"mx":245,"my":47}];
 $('.sigPad').signaturePad(options).regenerate(sig)
 })
 </script>
  
 If there isn't a signature drawn, I see this:
 <script>
 $(document).ready(function(){
var options = {
 bgColour : '#ffffff',
 drawOnly : true,
 penColour: '#0087bb',
 lineTop: 65,
 };
 var sig = ;
 $('.sigPad').signaturePad(options).regenerate(sig)
 })
 </script>
  
 I'll see if I can find the example files and try what you suggest. I guess I'm not really sure how this is used outside of my use in FileBound.
  
 Regards,
 Bill
 
 
--------- Original Message --------- Subject: Re: [thomasjbradley/signature-pad] How to make siganature-pad totally optional on a form (#182)

**Comment by radman63** [comment]
OK. I fixed it.You won't believe this. First, I used your sample files, making different versions of your sample-signature-output.js file to represent both a filled-in and a blank signature. I realized that what was really throwing the error was that when the signature pad is blank, it populated the value as a semi-colon, which "it" didn't like. I realized the semi-colon was just coming from the end of that line of script, so I just removed the semi-colon from the end of the line, and now everything works great, whether a signature is drawn or not.

In other words, I did this:

I changed  var sig = $$output$$; to  var sig = $$output$$ (removed the semi-colon from the end of the line)

Crazy, huh?

Thanks for your help.
